promql/promqltest/testdata/functions.test

```diff
@@ -234,11 +234,25 @@ clear
 load 5m
 	http_requests{path="/foo"}	0 50 100 150 200
 	http_requests{path="/bar"}	200 150 100 50 0
+	http_requests_gauge{path="/foo"} {{schema:0 sum:0 count:0 buckets:[0 0 0] counter_reset_hint:gauge}}+{{schema:0 sum:1 count:2 buckets:[1 1 1] counter_reset_hint:gauge}}x5
+	http_requests_counter{path="/foo"} {{schema:0 sum:0 count:0 buckets:[0 0 0]}}+{{schema:0 sum:1 count:2 buckets:[1 1 1]}}x5
+	http_requests_mix{path="/foo"} 0 50 100 {{schema:0 sum:0 count:0 buckets:[0 0 0]}} {{schema:0 sum:1 count:2 buckets:[1 1 1]}}
```

Comment:
Let's add `counter_reset_hint:gauge` here, too, so that we can be confident the warning is really triggered by the mix if floats and histograms (and not because the histogram is a counter histogram).

Reply:
Done!